primitives/src/migrate.rs

```diff
@@ -21,6 +21,15 @@ use frame_support::storage::unhashed::kill_prefix;
 use frame_support::{ReversibleStorageHasher, StorageHasher, Twox128};
 use sp_std::vec::Vec;
 
+/// A migration error type.
+#[derive(Clone, PartialEq, Eq, Encode, Decode, Debug)]
+pub enum MigrationError<T: Clone + PartialEq + Eq + Encode + Decode> {
```

Comment:
Why are the bounds here necessary?

Reply:
Yep, they are required by substrate as this type is exposed as an Event.

Reply:
That seems strange to me. All that should be needed is that `T: Clone` implies `MigrationError<T>: Clone` for example, not that `T: Clone` in general. What happens when you remove `Clone` here? What's the error?

Reply:
Fixed.